File: src/main/java/me/chell/samsara/impl/widget/Watermark.java
package me.chell.samsara.impl.widget;

import me.chell.samsara.Samsara;
import me.chell.samsara.api.util.Color;
import me.chell.samsara.api.value.Value;
import me.chell.samsara.api.widget.Widget;

public class Watermark extends Widget {

    public Watermark() {
        super("Watermark", 2, 2, 5, 5);
        width = getFontRenderer().getStringWidth(Samsara.NAME + " " + Samsara.VERSION);
        height = getFontRenderer().FONT_HEIGHT;
    }

    private final Value<ColorMode> colorMode = builder("Color", ColorMode.PRIMARY).build();
    private final Value<Color> customColor = builder("Value", Color.WHITE()).visible(b -> colorMode.getValue() == ColorMode.CUSTOM).build();
    private final Value<Boolean> shadow = builder("Shadow", true).build();

    @Override
    public void render() {
        drawString(Samsara.NAME + " " + Samsara.VERSION, getX(), getY(), width, shadow.getValue(), Align.LEFT, colorMode.getValue(), customColor.getValue().getARGB());
    }
}

File: src/main/kotlin/me/chell/samsara/impl/widget/Watermark.kt
package me.chell.samsara.impl.widget

import me.chell.samsara.api.feature.Widget
import me.chell.samsara.impl.gui.click.Button
import net.minecraft.client.util.math.MatrixStack

class Watermark: Widget("Watermark") {

    override fun load() {
        super.load()

        width = fontRenderer.getWidth("$MODNAME $MODVER").toDouble()
        height = fontRenderer.fontHeight.toDouble()
    }

    override fun render(matrices: MatrixStack, mouseX: Double, mouseY: Double, tickDelta: Float) {
        super.render(matrices, mouseX, mouseY, tickDelta)
        if(mouseX < 0 && !pinned.value) return

        fontRenderer.drawWithShadow(matrices, MODNAME, x.value.toFloat(), y.value.toFloat(), Button.primaryText.color.argb)
        fontRenderer.drawWithShadow(matrices, MODVER, x.value.toFloat() + fontRenderer.getWidth("$MODNAME "), y.value.toFloat(), Button.secondaryText.color.argb)
    }
}